context/app/static/js/components/search/Facets/DateRangeFacet.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { useCallback, useState, useEffect } from 'react';
+import Stack from '@mui/material/Stack';
+import { DatePicker, DatePickerProps } from '@mui/x-date-pickers/DatePicker';
+import { DateValidationError } from '@mui/x-date-pickers/models';
+import { format } from 'date-fns/format';
+
+import { useSearch } from '../Search';
+import { isDateFilter, useSearchStore } from '../store';
+import { getFieldLabel } from '../fieldConfigurations';
+import FacetAccordion from './FacetAccordion';
+
+interface DateRangeFacetProps {
+  field: string;
+}
+
+function DatePickerComponent({
+  minDate,
+  maxDate,
+  ...rest
+}: DatePickerProps<Date> & Required<Pick<DatePickerProps<Date>, 'minDate' | 'maxDate'>>) {
+  const [error, setError] = React.useState<DateValidationError | null>(null);
+
+  const errorMessage = React.useMemo(() => {
+    switch (error) {
+      case 'maxDate':
+        return `Please select a date of ${format(maxDate, 'MMMM yyyy')} or before.`;
+
+      case 'minDate': {
+        return `Please select a date of ${format(minDate, 'MMMM yyyy')} or after.`;
+      }
+
+      case 'invalidDate': {
+        return 'The date is not valid.';
+      }
+
+      default: {
+        return '';
+      }
+    }
+  }, [error, minDate, maxDate]);
+
+  return (
+    <DatePicker
+      sx={(theme) => ({
+        '.MuiOutlinedInput-input, .MuiInputLabel-root, .MuiMonthCalendar-root': {
+          fontSize: theme.typography.subtitle2.fontSize,
+        },
+      })}
+      views={['month', 'year']}
+      onError={(newError) => setError(newError)}
```

Comment:
This should help avoid any potential callback rerender spookiness since state setters are stable functions:

```suggestion
      onError={setError}
```